### PR TITLE
SMO SDK samples

### PR DESCRIPTION
**Pull request by AndyYuMS**
.net40 uses scripting as example
.net core uses configuring sql server setting as example
open powershell uses remote connection and iteration as example
dlls have been removed

### CODE REVIEW

File: doc/smocodesample/net40app/net40app.csproj [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="Microsoft.SqlServer.AzureStorageEnum, Version=13.100.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91, processorArchitecture=MSIL">
+      <HintPath>..\packages\SharedManagementObjects.140.1.9\lib\net40\Microsoft.SqlServer.AzureStorageEnum.dll</HintPath>
```
Comment (AndyYuMS) [inline doc/smocodesample/net40app/net40app.csproj]: Nuget package will be installed aside the project in a "packages" folder.

File: doc/smocodesample/CoreApp/ModifySetting/project.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "version": "1.0.0-*",
+  "buildOptions": {
+    "debugType": "portable",
+    "emitEntryPoint": true
+  },
+  "dependencies": {
+    "sharedmanagementobjects": "(140.0.0,150.0.0)"
```
Comment (kburtram) [inline doc/smocodesample/CoreApp/ModifySetting/project.json]: >140.0.0,150.0.0 [](start = 33, length = 15)

I believe there is a syntax that specifies any version of 140.0.0, but not 150.0.0.  I think that's the syntax we want to use when referencing the 140 SMO SDK.

File: doc/smocodesample/net40app/Scripting/Program.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlServer.Management.Sdk.Sfc;
+
+namespace Microsoft.SqlServer.Management.SmoSdkSamples
+{
+    // This application demostrates how to script out a sample database without dependencies and iterate through the list to display the results.
```
Comment (kburtram) [inline doc/smocodesample/net40app/Scripting/Program.cs]: this is still an issue.

---
In reply to: [88115682](https://github.com/Microsoft/sqltoolsservice/pull/130#discussion_r88115682) [](ancestors = 88115682)

File: doc/smocodesample/net40app/Scripting/Program.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+using Microsoft.SqlServer.Management.Sdk.Sfc;
+
+namespace Microsoft.SqlServer.Management.SmoSdkSamples
+{
+    // This application demonstrates how to script out a sample database without dependencies and iterate through the list to display the results.
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            //Microsoft.SqlServer.Management.Smo.Server
```
Comment (kburtram) [inline doc/smocodesample/net40app/Scripting/Program.cs]: >//Microsoft.SqlServer.Management.Smo.Server [](start = 12, length = 43)

what is this comment for?

File: doc/smocodesample/net40app/Scripting/Scripting.csproj [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="Microsoft.SqlServer.Management.Sdk.Sfc, Version=14.000.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91, processorArchitecture=MSIL">
```
Comment (kburtram) [inline doc/smocodesample/net40app/Scripting/Scripting.csproj]: >000 [](start = 75, length = 3)

why the "000"?  isn't the more typical way to write this "14.0.0.0"?

File: doc/smocodesample/net40app/Scripting/Program.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            // Connect to the local, default instance of SQL Server.
+            Smo.Server srv = new Smo.Server();
+	        // database name
+	        Console.WriteLine("Enter database name for scripting:"); 
+            string dbName = Console.ReadLine(); 
+            // Reference the database.    
+            Database db = srv.Databases[dbName];
+            // Define a Scripter object and set the required scripting options.   
+            Scripter scrp = new Scripter(srv);
```
Comment (kburtram) [inline doc/smocodesample/net40app/Scripting/Program.cs]: >scrp [](start = 21, length = 4)

can we name this variable "scripter"?

File: doc/smocodesample/net40app/Scripting/Scripting.csproj [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\packages\SharedManagementObjects.140.1.9\lib\net40\Microsoft.SqlServer.Management.Sdk.Sfc.dll</HintPath>
```
Comment (kburtram) [inline doc/smocodesample/net40app/Scripting/Scripting.csproj]: >..\packages [](start = 16, length = 11)

is this path going to be valid when I build the csproj file?  I still don't understand how open the CSProj file will create a packages folder in the parent directory?

File: doc/smocodesample/PowerShellScript/dbiteration.ps1 [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+$srv.ConnectionContext.Password = "pwd"
+
+$d = $srv.EnumCollations()
+
+Foreach ($r in $d.Rows)
+{
+   Write-Host "============================================"
+   Foreach ($c in $r.Table.Columns)
+   {
+      Write-Host $c.ColumnName "=" $r[$c].ToString()
```
Comment (kburtram) [inline doc/smocodesample/PowerShellScript/dbiteration.ps1]: >$r[$c] [](start = 35, length = 6)

can we use more descriptive variable names like $row and $columnName?

File: doc/smocodesample/CoreApp/project.json [hunk 9]
```diff
@@ -0,0 +1,21 @@
+{
+  "version": "1.0.0-*",
+  "buildOptions": {
+    "debugType": "portable",
+    "emitEntryPoint": true
+  },
+  "dependencies": {
+    "SharedManagementObjects":  "140.1.9"
```
Comment (ShaochenYu) [inline doc/smocodesample/CoreApp/project.json]: Done

File: doc/smocodesample/CoreApp/Program.cs [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+                Console.WriteLine("There are " + srv.Configuration.Properties.Count.ToString() + " configuration options.");
+                //Display the maximum and minimum values for ShowAdvancedOptions.   
+                int min = 0;
+                int max = 0;
+                min = srv.Configuration.ShowAdvancedOptions.Minimum;
+                max = srv.Configuration.ShowAdvancedOptions.Maximum;
+                Console.WriteLine("Minimum and Maximum values are " + min + " and " + max + ".");
+                //Modify the value of ShowAdvancedOptions and run the Alter method.   
+                srv.Configuration.ShowAdvancedOptions.ConfigValue = 0;
+                srv.Configuration.Alter();
```
Comment (ShaochenYu) [inline doc/smocodesample/CoreApp/Program.cs]: Done

File: doc/smocodesample/CoreApp/Program.cs [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+                //Display all the configuration options.   
+                foreach (ConfigProperty p in srv.Configuration.Properties)
+                {
+                    Console.WriteLine(p.DisplayName);
+                }
+                Console.WriteLine("There are " + srv.Configuration.Properties.Count.ToString() + " configuration options.");
+                //Display the maximum and minimum values for ShowAdvancedOptions.   
+                int min = 0;
+                int max = 0;
+                min = srv.Configuration.ShowAdvancedOptions.Minimum;
```
Comment (ShaochenYu) [inline doc/smocodesample/CoreApp/Program.cs]: Done

File: doc/smocodesample/CoreApp/Program.cs [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+                Server srv = new Server();
+                //Display all the configuration options.   
+                foreach (ConfigProperty p in srv.Configuration.Properties)
+                {
+                    Console.WriteLine(p.DisplayName);
+                }
+                Console.WriteLine("There are " + srv.Configuration.Properties.Count.ToString() + " configuration options.");
+                //Display the maximum and minimum values for ShowAdvancedOptions.   
+                int min = 0;
+                int max = 0;
```
Comment (ShaochenYu) [inline doc/smocodesample/CoreApp/Program.cs]: Done

File: doc/smocodesample/net40app/app.config [hunk 13]
```diff
@@ -0,0 +1,23 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="Microsoft.SqlServer.BatchParser" publicKeyToken="89845dcd8080cc91" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-13.100.0.0" newVersion="13.100.0.0" />
+      </dependentAssembly>
```
Comment (ShaochenYu) [inline doc/smocodesample/net40app/app.config]: Done

File: doc/smocodesample/net40app/net40app.csproj (hunk 1 above)
Comment (ShaochenYu) [inline doc/smocodesample/net40app/net40app.csproj]: Done

File: doc/smocodesample/net40app/Program.cs [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+{
+    // This application demostrates how to script out a sample database without dependencies and iterate through the list to display the results.
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            //Connect to the local, default instance of SQL Server.
+            Server srv = new Server();
+	    // database name 
+            string dbName = "AdventureWorks2012";
```
Comment (ShaochenYu) [inline doc/smocodesample/net40app/Program.cs]: use master, done

File: doc/smocodesample/net40app/net40app.csproj [hunk 15]
```diff
@@ -0,0 +1,199 @@
...
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectGuid>{C0CACFBB-FDAF-4B3D-B9F7-EBC3179A5055}</ProjectGuid>
+    <OutputType>Exe</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>net40app</RootNamespace>
+    <AssemblyName>net40app</AssemblyName>
```
Comment (ShaochenYu) [inline doc/smocodesample/net40app/net40app.csproj]: Done

**Pull request closed without merging**